pkg/controller-gen/generators/type_go.go

```diff
@@ -200,14 +205,38 @@ func (a *{{.lowerName}}GeneratingHandler) Handle(obj *{{.version}}.{{.type}}, st
 	}
 
 	objs, newStatus, err := a.{{.type}}GeneratingHandler(obj, status)
-	if err != nil {
+	if err != nil || !a.isNewResourceVersion(obj) {
 		return newStatus, err
 	}
 
-	return newStatus, generic.ConfigureApplyForObject(a.apply, obj, &a.opts).
+	err = generic.ConfigureApplyForObject(a.apply, obj, &a.opts).
 		WithOwner(obj).
 		WithSetID(a.name).
 		ApplyObjects(objs...)
+	if err == nil {
+		a.seenResourceVersion(obj)
+	}
+	return newStatus, err
```

Comment:
Because this code is written in a template and not actually go code, We should keep logic simple even if it costs an extra line.
```suggestion
	if err != nil {
		return newStatus, err
	}
	a.seenResourceVersion(obj)
	return newStatus, nil
```